### PR TITLE
Remove params from message ctors

### PR DESCRIPTION
**Pull request by JamesNK**
Removed `params` in message ctors so named arguments are no longer required to call.

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs [hunk 1]
```diff
@@ -294,8 +294,7 @@ private static void Complete(ConnectionContext connection)
             Log.PreparingBlockingInvocation(_logger, irq.InvocationId, methodName, irq.ResultType.FullName, args.Length);
 
             // Client invocations are always blocking
-            var invocationMessage = new InvocationMessage(irq.InvocationId, target: methodName,
-                argumentBindingException: null, arguments: args);
+            var invocationMessage = new InvocationMessage(irq.InvocationId, methodName, null, args);
```
[the remark this answers is not in the document]
Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: I hate that whole thing... I wish we didn't have it at all, it's gross to put an exception into the message itself...

[the remark this answers is not in the document]
Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: I'd prefer throwing inline. It's an error in parsing the message if you can't bind the arguments.

[the remark this answers is not in the document]
Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: Might have been different in protobuf, that might explain that

[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: > Why did you rename the named arguments?

I didn't rename the arguments?

The only reason they were there is passing null plus `params object[]` would collide, forcing them to be required.

File: benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs [hunk 2]
```diff
@@ -37,16 +37,16 @@ public void GlobalSetup()
             switch (Input)
             {
                 case Message.NoArguments:
-                    _hubMessage = new InvocationMessage(target: "Target", argumentBindingException: null);
+                    _hubMessage = new InvocationMessage("Target", null, Array.Empty<object>());
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs]: Hmmm... if it isn't obvious then I would prefer some static helper methods for creating messages over using named arguments everywhere. That is what `CompletionMessage` does:
`CompletionMessage.WithResult("invokeId", "a result")` vs `CompletionMessage.WithError("invokeId", "an error")`

[the remark this answers is not in the document]
Comment (anurse) [reply benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs]: If we get rid of `argumentBindingException` we don't have to worry about it at all... I guess leave it for now.

File: test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/MessagePackHubProtocolTests.cs [hunk 3]
```diff
@@ -58,175 +58,175 @@ public static IEnumerable<object[]> TestDataNames
             // Invocation messages
             new ProtocolTestData(
                 name: "InvocationWithNoHeadersAndNoArgs",
-                message: new InvocationMessage(invocationId: "xyz", target: "method", argumentBindingException: null),
+                message: new InvocationMessage("xyz", "method", null, System.Array.Empty<object>()),
```
[the remark this answers is not in the document]
Comment (anurse) [reply test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/MessagePackHubProtocolTests.cs]: What specific type/identifier is it conflicting with? I don't see any MessagePack in the `using`s...

[the remark this answers is not in the document]
Comment (JamesNK) [reply test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/MessagePackHubProtocolTests.cs]: Ah, it wasn't an Array type in msgpack, it is an Array method on the test class.

File: src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (moozzyk) [reply src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: It killed the connection instead of sending an error to the user. #815

**Comment by davidfowl** [comment]
I think we can drop it completely then. We now have a close message that will handle these cases.

File: src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: Looking into removing/actually removing issue - https://github.com/aspnet/SignalR/issues/1941

**Pull request merged**